chainer/link.py

```diff
@@ -172,27 +240,16 @@ def add_param(self, name, shape=None, dtype=numpy.float32,
                 ignored.
 
         """
-        d = self.__dict__
-        if name in d:
-            raise AttributeError(
-                'cannot register a new parameter %s: attribute exists'
-                % name)
```

Comment:
Is it possible to add test cases for the duplicated registration of `add_persistent` and `add_param`?